code-server/run.sh

```diff
@@ -27,7 +27,11 @@ fi
 
 # Check if code-server is already installed for offline or cached mode
 if [ -f "$CODE_SERVER" ]; then
-  if [ "${OFFLINE}" = true ] || [ "${USE_CACHED}" = true ]; then
+  if [ "${USE_CACHED}" = true ] && [ -n "${EXTENSIONS_DIR}" ] && [ ! -d "${EXTENSIONS_DIR}" ]; then
```

Comment:
I think it would be nice to describe the behavior, maybe in the use_cached var description.  Something like this maybe:

> "Uses cached copy code-server in the background, otherwise fetched it from GitHub. Additionally, if extensions_dir is set, auto_install_extensions will only run if that directory does not already exist."

Reply:
Oh I guess `extensions` is also involved.

> "Uses cached copy code-server in the background, otherwise fetched it from GitHub. Additionally, if extensions_dir is set, extensions will only be installed if that directory does not already exist."

Reply:
@code-asher  - sorry i made a further simplication.